File: mpp/Lab3/C#/model/Participant.cs
public class Participant : Entity<long>
{
    public string Nume { get; set; }
    public string Prenume { get; set; }
    public long Varsta { get; set; }

    public Participant(long id, string nume, string prenume, long varsta)
    {
        Id = id;
        Nume = nume;
        Prenume = prenume;
        Varsta = varsta;
    }

    public override string ToString()
    {
        return "Participant{" +
                "id=" + Id +
                ", nume='" + Nume + '\'' +
                ", prenume='" + Prenume + '\'' +
                ", varsta='" + Varsta + '\'' +
                '}';
    }
}

File: mpp/Lab3/java/model/Participant.java
package ro.mpp2024.model;

import java.util.Objects;

public class Participant extends Entity<Long> {
    private String nume,prenume;
    private int varsta;

    public Participant() {
    }

    public Participant(Long id,String nume, String prenume, int varsta) {
        setId(id);
        this.nume = nume;
        this.prenume = prenume;
        this.varsta = varsta;
    }

    public String getNume() {
        return nume;
    }

    public void setNume(String nume) {
        this.nume = nume;
    }

    public String getPrenume() {
        return prenume;
    }

    public void setPrenume(String prenume) {
        this.prenume = prenume;
    }

    public int getVarsta() {
        return varsta;
    }

    public void setVarsta(int varsta) {
        this.varsta = varsta;
    }

    @Override
    public String toString() {
        return "Participant{" +
                "nume='" + nume + '\'' +
                ", prenume='" + prenume + '\'' +
                ", varsta=" + varsta +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Participant participant)) return false;

        if (!Objects.equals(nume, participant.nume)) return false;
        if (!Objects.equals(prenume, participant.prenume)) return false;
        return varsta == participant.varsta;
    }
}
